### DISCUSSION TITLE
Deleting file after sending using SystemFile

### DISCUSSION THREAD

**Discussion by morki** [micronaut-projects/micronaut-core#4868] [Q&A] [answered]
Hi, can you please help me in solving following situation? 

I am generation ZIP reporting file in controller action and sending it using `SystemFile` to client. But this file is only temporary and need to be deleted after sending. Is this supported somehow? Or is there any workaround?

Example from documentation:
```kotlin
@Get
public SystemFile download() {
    File file = ...
    return new SystemFile(file).attach("myfile.txt");
}
```

Thank you very much for any help :)

**Comment by jameskleeh** [comment, marked as answer]
@morki There isn't an API to do that today. You could likely return a StreamedFile and pass in a custom inputstream that overrides the close method. You can create an issue to ask for a new feature to support this

**Reply by morki** [reply]
That you this solution is totally fine for me and I dont think it has to be in framework itseft, thank you very much :)
